lib/practice_exercises.rb

```diff
@@ -1,13 +1,33 @@
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n)
+# Space Complexity: O(n)
 def is_palindrome(string)
-  raise NotImplementedError, "Not implemented yet"
+  return true if string.empty? || string.length == 1
+  string.downcase!
+  split = string.split("")
+  characters = split.select{ |char| /[a-z]/.match?(char) }
+
+  mid = characters.length / 2
+  mid.times do |i|
+    return false if characters[i] != characters[(characters.length - 1) - i]
+  end
+
+  return true
 end
 
-# Time Complexity: ?
-# Space Complexity: ?
+# Time Complexity: O(n^2)
+# Space Complexity: O(1)
 def longest_prefix(strings)
```

Comment:
This has 2 loops, the outer goes through the length of the 1st string (m), and the inner goes through the list of strings (n).  So the time complexity is O(nm).

It's building a string `longest` so the space complexity is O(m)